src/llmq/quorums_instantsend.cpp

```diff
@@ -963,62 +963,29 @@ void CInstantSendManager::ProcessInstantSendLock(NodeId from, const uint256& has
     }
 }
 
-void CInstantSendManager::ProcessNewTransaction(const CTransactionRef& tx, const CBlockIndex* pindex, bool allowReSigning)
-{
-    if (!IsInstantSendEnabled()) {
-        return;
-    }
-
-    if (tx->IsCoinBase() || tx->vin.empty()) {
-        // coinbase and TXs with no inputs can't be locked
-        return;
-    }
-
-    uint256 islockHash;
-    {
-        LOCK(cs);
-        islockHash = db.GetInstantSendLockHashByTxid(tx->GetHash());
-
-        // update DB about when an IS lock was mined
-        if (!islockHash.IsNull() && pindex) {
-            db.WriteInstantSendLockMined(islockHash, pindex->nHeight);
-        }
-    }
-
-    if (!masternodeSync.IsBlockchainSynced()) {
-        return;
-    }
-
-    bool chainlocked = pindex && chainLocksHandler->HasChainLock(pindex->nHeight, pindex->GetBlockHash());
-    if (islockHash.IsNull() && !chainlocked) {
-        ProcessTx(*tx, allowReSigning, Params().GetConsensus());
-    }
-
-    LOCK(cs);
-    if (!chainlocked && islockHash.IsNull()) {
-        // TX is not locked, so make sure it is tracked
-        AddNonLockedTx(tx, pindex);
-    } else {
-        // TX is locked, so make sure we don't track it anymore
-        RemoveNonLockedTx(tx->GetHash(), true);
-    }
-}
-
 void CInstantSendManager::TransactionAddedToMempool(const CTransactionRef& tx)
 {
-    if (!IsInstantSendEnabled()) {
+    if (!IsInstantSendEnabled() || !masternodeSync.IsBlockchainSynced() || tx->vin.empty()) {
         return;
     }
 
-    CInstantSendLockPtr islock{nullptr};
+    CInstantSendLockPtr islock;
```

Comment:
why remove the `{nullptr}`?

Reply:
It's assigned right below unconditionally, so there is no need to initialize it here and it should be safe. But it wasn't intentional actually, was just smth I did while moving things around. Let's revert this.

```suggestion
    CInstantSendLockPtr islock{nullptr};
```